source/core/src/main/com/csse3200/game/components/minigames/birdieDash/BirdieDashGame.java

```diff
@@ -1,3 +1,166 @@
+    private Boolean isGameOver;
+
+    public BirdieDashGame() {
+        this.pipes = createPipes();
+        this.coins = createCoins();
+        this.bird = new Bird(920, 600);
+        this.spike = new Spike(0);
+        this.renderer = new MinigameRenderer();
+        this.isGameOver = false;
+      //  this.controller = new BirdieDashController();
+        this.collisionHandler = new CollisionHandler(bird, pipes, coins, spike);
+        initRenderers();
+    }
+
+    /**
+     * Private method to set up game renderers
+     */
+    private void initRenderers() {
+        ServiceLocator.registerResourceService(new ResourceService());
+        renderer.addRenderable(new BackgroundRenderer(renderer));
+        renderer.addRenderable(new PipeRenderer(pipes, renderer));
+        renderer.addRenderable(new CoinRenderer(coins, renderer));
+        renderer.addRenderable(new BirdRenderer(bird, renderer));
+        renderer.addRenderable(new SpikeRenderer(spike, renderer));
+        ServiceLocator.getResourceService().loadTextures(new String[]{"images/PauseOverlay/TitleBG.png",
+                "images/PauseOverlay/Button.png",
+                "images/QuestsOverlay/Quest_BG.png",
+                "images/QuestsOverlay/Quest_SBG.png"});
+        ServiceLocator.getResourceService().loadAll();
+    }
+
+    /**
+     * Private method to create a list of coins
+     * @return the list of coins
+     */
+    private List<Coin> createCoins() {
+        List<Coin> coins = new ArrayList<>();
+        for(int i= 0; i < 3; i++) {
+            coins.add(new Coin(1920 + 960 * i, START_SPEED));
+        }
+        return coins;
+    }
+
+    /**
+     * Change the coins position
+     * @param dt time since last coin position
+     */
+    private void changeCoinPosition(float dt) {
+        for(Coin coin : coins) {
+            coin.changePosition(dt);
+        }
+    }
+
+    /**
+     * Private method to create pipes for the start of the game
+     * @return a list containing all the pipe objects to be used.
+     */
+    private List<Pipe> createPipes() {
+        List<Pipe> pipes = new ArrayList<>();
+        for(int i = 0; i < 3; i++) {
+            // Only need three pipes
+            pipes.add(new Pipe(1440 + 960 * i, START_SPEED));
+        }
+        return pipes;
+    }
+
+    /**
+     * Private method to change all the pipe's positions based on the change in
+     * game time
+     * @param dt the change in game time
+     */
+    private void changePipePosition(float dt) {
+        for(Pipe pipe : pipes) {
+            pipe.changePosition(dt);
+        }
+    }
+
+    /**
+     * Private method to update all positions of entities in the game
+     * @param dt the change in game time
+     */
+    private void updateGamePosition(float dt) {
+        speedMultiplier += accelerationRate * dt;
+        collisionHandler.checkCollisions();
+        if (collisionHandler.checkSpikes() || bird.touchingFloor()) {
+            isGameOver = true;
```

Comment:
just a suggestion :) maybe handle game over in a separate method so this just updates game positions only